.github/workflows/github-actions-tests.yml

```
@@ -7,22 +7,18 @@ jobs:
       fail-fast: false
       matrix:
         python-version: [ "3.9", "3.10", "3.11", "3.12" ]
-        poetry-version: [ 1.2 ]
-        pydantic-version: [ 1.10.0, 2.11 ]
         os: [ ubuntu-latest ]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ inputs.python-version }}
+        uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Run image
-        uses: abatilo/actions-poetry@v2
+          python-version: ${{ inputs.python-version }}
```

Same as previous comment, otherwise the correct Python version wouldn't be picked up by the CI.
```suggestion
          python-version: ${{ matrix.python-version }}
```